### PR TITLE
Replace "contract instance" with "contract" in docs folder

### PR DESCRIPTION
**Pull request by richardkapolnai-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/concepts/glossary.rst [hunk 1]
```diff
@@ -9,7 +9,7 @@ DAML
 
 **DAML** is a programming language for writing `smart contracts <#contract-contract-instance>`__, that you can use to build an application based on a `ledger <#ledger-daml-ledger>`__. You can run DAML contracts on many different ledgers.
 
-Contract, contract instance
+Contract, contract
```
Comment (richardkapolnai-da) [inline docs/source/concepts/glossary.rst]: remove contract instance
```suggestion
Contract
```

File: docs/source/concepts/glossary.rst [hunk 2]
```diff
@@ -20,7 +20,7 @@ A **contract** is an item on a `ledger <#ledger-daml-ledger>`__. They are create
 
 Contracts are immutable: once they are created on the ledger, the information in the contract cannot be changed. The only thing that can happen to it is that the contract can be `archived <#active-contract-archived-contract>`__.
 
-They're sometimes referred to as a **contract instance** to make clear that this is an instantiated contract, as opposed to a `template <#template>`__.
+They're sometimes referred to as a **contract** to make clear that this is an instantiated contract, as opposed to a `template <#template>`__.
```
Comment (richardkapolnai-da) [inline docs/source/concepts/glossary.rst]: They should not be referred as such.
```suggestion

```

File: docs/source/concepts/glossary.rst [hunk 3]
```diff
@@ -20,8 +20,6 @@ A **contract** is an item on a `ledger <#ledger-daml-ledger>`__. They are create
 
 Contracts are immutable: once they are created on the ledger, the information in the contract cannot be changed. The only thing that can happen to it is that the contract can be `archived <#active-contract-archived-contract>`__.
 
-They're sometimes referred to as a **contract instance** to make clear that this is an instantiated contract, as opposed to a `template <#template>`__.
```
Comment (richardkapolnai-da) [inline docs/source/concepts/glossary.rst]: They should not be referred as such, so I removed this note.

File: docs/source/concepts/glossary.rst [hunk 4]
```diff
@@ -70,7 +68,7 @@ A `choice <#choice>`__ marked **postconsuming** will not be `archived <#active-c
 Nonconsuming choice
 --------------------
 
-A **nonconsuming choice** does NOT `archive <#active-contract-archived-contract>`__ the `contract <#contract-contract-instance>`__ it is on when `exercised <#exercise>`__. This means the choice can be exercised more than once on the same `contract instance <#contract-contract-instance>`__. 
+A **nonconsuming choice** does NOT `archive <#active-contract-archived-contract>`__ the `contract <#contract-contract-instance>`__ it is on when `exercised <#exercise>`__. This means the choice can be exercised more than once on the same `contract <#contract-contract-instance>`__.
```
Comment (richardkapolnai-da) [inline docs/source/concepts/glossary.rst]:
```suggestion
A **nonconsuming choice** does NOT `archive <#active-contract-archived-contract>`__ the `contract <#contract-contract-instance>`__ it is on when `exercised <#exercise>`__. This means the choice can be exercised more than once on the same `contract <#contract>`__. 
```

File: docs/source/concepts/ledger-model/ledger-daml.rst [hunk 5]
```diff
@@ -51,7 +51,7 @@ Intuitively, the allowed actions are:
 
 #. **NoSuchKey** assertions corresponding to a :ref:`lookupByKey` update statement for the given key that does not find a contract.
 
-An instance of a DAML template, that is, a **DAML contract** or **contract instance**,
+An instance of a DAML template, that is, a **DAML contract** or **contract**,
```
Comment (richardkapolnai-da) [inline docs/source/concepts/ledger-model/ledger-daml.rst]:
```suggestion
An instance of a DAML template, that is, a **DAML contract**,
```

**Comment by richardkapolnai-da** [comment]
Please see also internal conversation on #product-daml.

File: docs/source/concepts/glossary.rst [hunk 6]
```diff
@@ -7,9 +7,9 @@ Glossary of concepts
 DAML
 ****
 
-**DAML** is a programming language for writing `smart contracts <#contract-contract-instance>`__, that you can use to build an application based on a `ledger <#ledger-daml-ledger>`__. You can run DAML contracts on many different ledgers.
+**DAML** is a programming language for writing `smart contracts <#contract>`__, that you can use to build an application based on a `ledger <#ledger-daml-ledger>`__. You can run DAML contracts on many different ledgers.
 
-Contract, contract instance
```
Comment (richardkapolnai-da) [inline docs/source/concepts/glossary.rst]: This is why I needed to change every ref from `contract-contract-instance` to `contract`

**Review by bame-da — APPROVED** [review]
Sorry, this seems to have slipped between the cracks entirely. If you still want to get this merged, I'm fine with it.

**Comment by richardkapolnai-da** [comment]
> Sorry, this seems to have slipped between the cracks entirely. If you still want to get this merged, I'm fine with it.

Thanks and no worries, I'm happy that it can be merged! Will do a bit later (also will resolv the conflicts).

Actually, as far as I remember, the conversation was also about eliminating the [usage of "contract template"](https://www.google.com/search?q=%22contract+template%22+site%3Adocs.daml.com). I created #7551 to remember that.

**Pull request merged**